src/common/utility.cpp

```diff
@@ -51,7 +51,7 @@ timeval ToTimeval(std::chrono::microseconds microseconds) {
 static void WriteEscapedString(std::ostringstream& writer,
                                opentracing::string_view s) {
   writer << '"';
-  for (char c : s) {
+  for (signed char c : s) {
     switch (c) {
```

Comment:
Could you change to
```
for (auto c : s) {
   ...
}
```

Reply:
With `auto` I'm still getting the same error:

```
  /usr/bin/aarch64-linux-gnu-gcc -fstack-protector -isystem /usr/aarch64-linux-gnu/include/c++/9 -isystem /usr/aarch64-linux-gnu/include/c++/9/aarch64-linux-gnu -isystem /usr/aarch64-linux-gnu/include/c++/9/backward -isystem /usr/lib/gcc-cross/aarch64-linux-gnu/9/include -isystem /usr/aarch64-linux-gnu/include -isystem /usr/include -MD -MF bazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/src/common/_objs/utility_lib/utility.pic.d '-frandom-seed=bazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/src/common/_objs/utility_lib/utility.pic.o' -fPIC -iquote external/com_lightstep_tracer_cpp -iquote bazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp -iquote external/com_google_protobuf -iquote bazel-out/k8-fastbuild/bin/external/com_google_protobuf -iquote external/io_opentracing_cpp -iquote bazel-out/k8-fastbuild/bin/external/io_opentracing_cpp -iquote external/com_google_googleapis -iquote bazel-out/k8-fastbuild/bin/external/com_google_googleapis -Ibazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/src/common/_virtual_includes/utility_lib -Ibazel-out/k8-fastbuild/bin/external/io_opentracing_cpp/_virtual_includes/opentracing -Ibazel-out/k8-fastbuild/bin/external/io_opentracing_cpp/3rd_party/_virtual_includes/expected -Ibazel-out/k8-fastbuild/bin/external/io_opentracing_cpp/3rd_party/_virtual_includes/variant -Ibazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/src/common/platform/_virtual_includes/time_lib -Ibazel-out/k8-fastbuild/bin/external/com_google_protobuf/_virtual_includes/descriptor_proto -Ibazel-out/k8-fastbuild/bin/external/com_google_protobuf/_virtual_includes/timestamp_proto -Ibazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/src/common/_virtual_includes/logger_lib -Ibazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/include/lightstep/_virtual_includes/tracer_interface -Ibazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/include/lightstep/_virtual_includes/transporter_interface -Ibazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/include/lightstep/_virtual_includes/buffer_chain_interface -Ibazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/include/lightstep/_virtual_includes/metrics_observer_interface -Ibazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/src/common/_virtual_includes/hex_conversion_lib -isystem external/com_google_protobuf/src -isystem bazel-out/k8-fastbuild/bin/external/com_google_protobuf/src -isystem bazel-out/k8-fastbuild/bin/external/envoy/bazel/foreign_cc/zlib/include -fPIC -Iexternal/com_lightstep_tracer_cpp/include -Ibazel-out/k8-fastbuild/binexternal/com_lightstep_tracer_cpp//include -Iexternal/com_lightstep_tracer_cpp/src -Ibazel-out/k8-fastbuild/binexternal/com_lightstep_tracer_cpp//src -Iexternal/com_lightstep_tracer_cpp/test -Ibazel-out/k8-fastbuild/binexternal/com_lightstep_tracer_cpp//test -isystem external/com_lightstep_tracer_cpp/3rd_party/base64/include -isystem bazel-out/k8-fastbuild/binexternal/com_lightstep_tracer_cpp//3rd_party/base64/include -Wall -Wextra -Werror -Wnon-virtual-dtor -Woverloaded-virtual -Wold-style-cast -Wno-overloaded-virtual -Wvla '-std=c++11' -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -no-canonical-prefixes -fno-canonical-system-headers -c external/com_lightstep_tracer_cpp/src/common/utility.cpp -o bazel-out/k8-fastbuild/bin/external/com_lightstep_tracer_cpp/src/common/_objs/utility_lib/utility.pic.o)
Execution platform: @local_config_platform//:host

Use --sandbox_debug to see verbose messages from the sandbox
external/com_lightstep_tracer_cpp/src/common/utility.cpp: In function 'void lightstep::WriteEscapedString(std::ostringstream&, opentracing::v2::string_view)':
external/com_lightstep_tracer_cpp/src/common/utility.cpp:75:20: error: comparison is always true due to limited range of data type [-Werror=type-limits]
   75 |         if ('\x00' <= c && c <= '\x1f') {
      |             ~~~~~~~^~~~
cc1plus: all warnings being treated as errors
Target //source/exe:envoy-static failed to build
INFO: Elapsed time: 13.964s, Critical Path: 1.70s
INFO: 1 process: 1 processwrapper-sandbox.
```

(I'm building [Envoy](https://github.com/envoyproxy/envoy) which uses lightstep-tracer-cpp as a dependency)